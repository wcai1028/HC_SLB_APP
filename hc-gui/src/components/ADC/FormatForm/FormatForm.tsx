import * as React from 'react'
import { A10Component } from 'a10-gui-framework'
import { A10Icon, A10Button } from 'a10-gui-widgets'

// tslint:disable:no-var-requires
const styles = require('./styles/index.module.less')

export interface IFormatFormProps {
  title?: string
  description?: string
  children: React.ReactElement<any> | Array<React.ReactElement<any>>
  showBack?: boolean
  saveText?: string
  disableSave?: boolean
  hideCancel?: boolean
  customizedButtons?: React.ReactElement<any>
  onRequestOk?: (isOpen: boolean) => void
  onRequestClose?: (isOpen: boolean) => void
}

class FormatForm extends A10Component<IFormatFormProps> {
  static defaultProps = {
    showBack: false,
    disableSave: false,
    hideCancel: false,
    onRequestOk: (isOpen: boolean) => {
      return
    },
    onRequestClose: (isOpen: boolean) => {
      return
    },
  }

  onClickSaveButton = () => {
    const { onRequestOk } = this.props
    onRequestOk(false)
  }

  onClickCancelButton = () => {
    const { onRequestClose } = this.props
    onRequestClose(false)
  }

  onClickBack = () => {
    const { onRequestClose } = this.props
    onRequestClose(false)
  }

  render() {
    const {
      title,
      description,
      children,
      showBack,
      saveText,
      disableSave,
      hideCancel,
      customizedButtons,
    } = this.props

    return (
      <>
        <div className={styles.a10SlidingPageContent}>
          <div className={styles.a10SlidingPageTitle}>{title}</div>
          <div className={styles.a10SlidingPageDescription}>{description}</div>
          {children}
        </div>
        <div className={styles.a10SlidingPageFooter}>
          {showBack ? (
            <a className="back-link" onClick={this.onClickBack}>
              <A10Icon type="double-left" /> Back
            </a>
          ) : null}
          <A10Button
            type="primary"
            className="submit-button"
            onClick={this.onClickSaveButton}
            disabled={disableSave}
          >
            {saveText ? saveText : 'Save'}
          </A10Button>
          {customizedButtons ? customizedButtons : null}
          {hideCancel ? null : (
            <A10Button className="cancel-button" onClick={this.onClickCancelButton}>
              Cancel
            </A10Button>
          )}
        </div>
      </>
    )
  }
}
export default FormatForm
