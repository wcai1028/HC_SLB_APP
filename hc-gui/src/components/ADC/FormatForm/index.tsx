export { default, default as FormatForm } from './FormatForm'
