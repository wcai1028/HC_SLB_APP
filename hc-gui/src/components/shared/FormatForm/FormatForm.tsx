import * as React from 'react'
import { A10Component } from 'a10-gui-framework'
import { A10Icon, A10Button } from 'a10-gui-widgets'

const styles = require('./styles/index.module.less')

export interface IFormatFormProps {
  title?: string
  description?: string
  children: React.ReactElement<any> | Array<React.ReactElement<any>>
  showBack?: boolean
  saveText?: string
  disableSave?: boolean
  hideCancel?: boolean
  alignSaveButtonToRight?: boolean
  customizedButtons?: React.ReactElement<any>
  onRequestOk?: (isOpen: boolean) => void
  onRequestClose?: (isOpen: boolean) => void
}

class FormatForm extends A10Component<IFormatFormProps> {
  static defaultProps = {
    onRequestOk: (isOpen: boolean) => {
      return
    },
  }

  closeSlidingPage = () => {
    const { onRequestClose } = this.props
    onRequestClose(false)
  }

  render() {
    const {
      title,
      description,
      children,
      showBack = false,
      saveText,
      disableSave,
      hideCancel,
      alignSaveButtonToRight,
      customizedButtons,
      onRequestOk,
    } = this.props
    return (
      <>
        <div className={styles.a10SlidingPageContent}>
          <div className={styles.a10SlidingPageTitle}>{title}</div>
          <div className={styles.a10SlidingPageDescription}>{description}</div>
          {children}
        </div>
        <div className={styles.a10SlidingPageFooter}>
          {showBack ? (
            <a className="back-link" onClick={this.closeSlidingPage}>
              <A10Icon type="double-left" /> Back
            </a>
          ) : null}
          {alignSaveButtonToRight ? (
            <A10Button
              type="primary"
              className="submit-button"
              onClick={onRequestOk}
              disabled={disableSave}
              style={{ float: 'right' }}
            >
              {saveText ? saveText : 'Save'}
            </A10Button>
          ) : (
            <A10Button
              type="primary"
              className="submit-button"
              onClick={onRequestOk}
              disabled={disableSave}
            >
              {saveText ? saveText : 'Save'}
            </A10Button>
          )}
          {customizedButtons ? customizedButtons : null}
          {hideCancel ? null : (
            <A10Button
              className="cancel-button"
              onClick={this.closeSlidingPage}
            >
              Cancel
            </A10Button>
          )}
        </div>
      </>
    )
  }
}
export default FormatForm
