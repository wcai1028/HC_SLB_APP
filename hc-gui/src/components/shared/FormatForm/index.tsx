export { default, default as FormatForm, IFormatFormProps } from './FormatForm'
