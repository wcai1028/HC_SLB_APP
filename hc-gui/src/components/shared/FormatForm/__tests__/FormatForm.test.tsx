import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'
import FormatForm from '../index'

describe('should FormatForm work', () => {
  it('should render properly', () => {
    const component = ReactTestRenderer.create(
      <FormatForm onRequestClose={jest.fn()}>
        <div>test</div>
      </FormatForm>,
    )
    const formatFormInstance = component.root.find(
      (node: any) => node.type.name === 'FormatForm',
    ).instance
    expect(component.toJSON()).toMatchSnapshot()
    formatFormInstance.closeSlidingPage()
    expect(formatFormInstance.props.onRequestClose).toHaveBeenCalled()
    expect(FormatForm.defaultProps.onRequestOk(true)).toBe(undefined)
  })

  it('should render properly with custom text for save button', () => {
    const component = ReactTestRenderer.create(
      <FormatForm
        showBack={true}
        saveText="Test"
        hideCancel={true}
        customizedButtons={<button>Test</button>}
      >
        <div>test</div>
      </FormatForm>,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly with aliignSaveButtonToRight without custom text for save button', () => {
    const component = ReactTestRenderer.create(
      <FormatForm
        showBack={true}
        alignSaveButtonToRight={true}
        hideCancel={true}
        customizedButtons={<button>Test</button>}
      >
        <div>test</div>
      </FormatForm>,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly with aliignSaveButtonToRight with custom text for save button', () => {
    const component = ReactTestRenderer.create(
      <FormatForm
        showBack={true}
        saveText="Test"
        alignSaveButtonToRight={true}
        hideCancel={true}
        customizedButtons={<button>Test</button>}
      >
        <div>test</div>
      </FormatForm>,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
