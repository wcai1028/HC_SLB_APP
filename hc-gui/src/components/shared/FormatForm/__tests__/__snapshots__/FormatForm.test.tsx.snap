// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should FormatForm work should render properly 1`] = `
Array [
  <div
    className="a10SlidingPageContent"
  >
    <div
      className="a10SlidingPageTitle"
    />
    <div
      className="a10SlidingPageDescription"
    />
    <div>
      test
    </div>
  </div>,
  <div
    className="a10SlidingPageFooter"
  >
    <button
      className="ant-btn submit-button ant-btn-primary ant-btn-lg"
      onClick={[Function]}
      type="button"
    >
      <span>
        Save
      </span>
    </button>
    <button
      className="ant-btn cancel-button ant-btn-lg"
      onClick={[Function]}
      type="button"
    >
      <span>
        Cancel
      </span>
    </button>
  </div>,
]
`;

exports[`should FormatForm work should render properly with aliignSaveButtonToRight with custom text for save button 1`] = `
Array [
  <div
    className="a10SlidingPageContent"
  >
    <div
      className="a10SlidingPageTitle"
    />
    <div
      className="a10SlidingPageDescription"
    />
    <div>
      test
    </div>
  </div>,
  <div
    className="a10SlidingPageFooter"
  >
    <a
      className="back-link"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-double-left"
      />
       Back
    </a>
    <button
      className="ant-btn submit-button ant-btn-primary ant-btn-lg"
      onClick={[Function]}
      style={
        Object {
          "float": "right",
        }
      }
      type="button"
    >
      <span>
        Test
      </span>
    </button>
    <button>
      Test
    </button>
  </div>,
]
`;

exports[`should FormatForm work should render properly with aliignSaveButtonToRight without custom text for save button 1`] = `
Array [
  <div
    className="a10SlidingPageContent"
  >
    <div
      className="a10SlidingPageTitle"
    />
    <div
      className="a10SlidingPageDescription"
    />
    <div>
      test
    </div>
  </div>,
  <div
    className="a10SlidingPageFooter"
  >
    <a
      className="back-link"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-double-left"
      />
       Back
    </a>
    <button
      className="ant-btn submit-button ant-btn-primary ant-btn-lg"
      onClick={[Function]}
      style={
        Object {
          "float": "right",
        }
      }
      type="button"
    >
      <span>
        Save
      </span>
    </button>
    <button>
      Test
    </button>
  </div>,
]
`;

exports[`should FormatForm work should render properly with custom text for save button 1`] = `
Array [
  <div
    className="a10SlidingPageContent"
  >
    <div
      className="a10SlidingPageTitle"
    />
    <div
      className="a10SlidingPageDescription"
    />
    <div>
      test
    </div>
  </div>,
  <div
    className="a10SlidingPageFooter"
  >
    <a
      className="back-link"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-double-left"
      />
       Back
    </a>
    <button
      className="ant-btn submit-button ant-btn-primary ant-btn-lg"
      onClick={[Function]}
      type="button"
    >
      <span>
        Test
      </span>
    </button>
    <button>
      Test
    </button>
  </div>,
]
`;
